plugins/modules/packaging/os/xbps.py

```diff
@@ -188,7 +223,12 @@ def install_packages(module, xbps_path, state, packages):
     cmd = "%s -y %s" % (xbps_path['install'], " ".join(toInstall))
     rc, stdout, stderr = module.run_command(cmd, check_rc=False)
 
-    if rc != 0 and not (state == 'latest' and rc == 17):
+    if rc == 16 and module.params['upgrade_xbps']:
+        upgrade_xbps(module, xbps_path)
+        # avoid loops by not trying self-update again
+        module.params['upgrade_xbps'] = False
+        install_packages(module, xbps_path, state, packages)
+    elif rc != 0 and not (state == 'latest' and rc == 17):
         module.fail_json(msg="failed to install %s" % (package))
 
     module.exit_json(changed=True, msg="installed %s package(s)"
```

Comment:
The next `module.exit_json()` a couple of lines below doesn't really make sense. I guess it can be deleted.

Reply:
done